### PR TITLE
Feature: Updated Chat History for Astra DB

### PR DESCRIPTION
**Pull request by erichare**
This Pull Request adds Chat History based on session IDs for Astra DB.

### CODE REVIEW

**Review by ogabrielluiz — APPROVED** [review]
LGTM

File: src/backend/base/langflow/components/memories/AstraDBMessageReader.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
+from typing import Optional, cast
+
+from langchain_community.chat_message_histories.astradb import AstraDBChatMessageHistory
```
Comment (nicoloboschi) [inline src/backend/base/langflow/components/memories/AstraDBMessageReader.py:3]: we should use the new package

**Review by nicoloboschi — CHANGES_REQUESTED** [review]

File: src/backend/base/langflow/components/memories/AstraDBMessageWriter.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                "display_name": "Astra DB Application Token",
+                "info": "Token for the Astra DB instance.",
+                "password": True,
+            },
+            "api_endpoint": {
+                "display_name": "Astra DB API Endpoint",
+                "info": "API Endpoint for the Astra DB instance.",
+                "password": True,
+            },
+            "namespace": {
```
Comment (nicoloboschi) [inline src/backend/base/langflow/components/memories/AstraDBMessageWriter.py:41]: "advanced": True, like the vector store

File: src/backend/base/langflow/components/memories/AstraDBMessageReader.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                "display_name": "Astra DB Application Token",
+                "info": "Token for the Astra DB instance.",
+                "password": True,
+            },
+            "api_endpoint": {
+                "display_name": "Astra DB API Endpoint",
+                "info": "API Endpoint for the Astra DB instance.",
+                "password": True,
+            },
+            "namespace": {
```
Comment (nicoloboschi) [inline src/backend/base/langflow/components/memories/AstraDBMessageReader.py:36]: "advanced": True, like the vector store

File: src/backend/base/langflow/components/memories/AstraDBMessageReader.py (hunk 1 above)
Comment (erichare) [reply src/backend/base/langflow/components/memories/AstraDBMessageReader.py]: Updated, good call

File: src/backend/base/langflow/components/memories/AstraDBMessageWriter.py (hunk 2 above)
Comment (erichare) [reply src/backend/base/langflow/components/memories/AstraDBMessageWriter.py:41]: Done.

File: src/backend/base/langflow/components/memories/AstraDBMessageReader.py (hunk 3 above)
Comment (erichare) [reply src/backend/base/langflow/components/memories/AstraDBMessageReader.py:36]: Done.

**Review by nicoloboschi — APPROVED** [review]
LGTM, great work!

**Comment by erichare** [comment]
@ogabrielluiz anything holding back the merge? I see a build issue but i think thats unrelated, correct me if im wrong?

I'd be happy to add a test or some kind of validation, just might need a little guidance there - maybe there's something for Zep available? Just let me know if so!

**Comment by ogabrielluiz** [comment]
@erichare should be good to go. It is still a bit hard to add automated tests for these integration components.

**Pull request merged**